exp/internal/controllers/machinepool_controller_phases.go

```diff
@@ -69,12 +69,20 @@ func (r *MachinePoolReconciler) reconcilePhase(mp *expv1.MachinePool) {
 
 	// Set the phase to "scalingUp" if the infrastructure is scaling up.
 	if mp.Status.InfrastructureReady && *mp.Spec.Replicas > mp.Status.ReadyReplicas {
-		mp.Status.SetTypedPhase(expv1.MachinePoolPhaseScalingUp)
+		if mp.Annotations[clusterv1.ReplicasManagedByAutoscalerAnnotation] == "true" {
```

Comment:
nit: instead of checking for true value, should this just be looking for the existence of the annotation?

Reply:
+1

Reply:
If just looking for existence can we at least also check that the value isn't specifically set to `false` please. I can already see a situation where this annotation was originally wanted and someone tried to then disable it by setting from `true` to `false` (such as when managing resources in a gitops approach)

Reply:
Something like:
```suggestion
		if ok, val := mp.Annotations[clusterv1.ReplicasManagedByAutoscalerAnnotation]; ok && val != "false" {
```

Reply:
Interesting point. We have a few of annotations like this and as far as I can see we usually only check for existence (`cluster.x-k8s.io/paused`, `cluster.x-k8s.io/skip-remediation`, `cluster.x-k8s.io/managed-by`).

I'm not really opposed to also checking that the annotation is not false, but I would prefer if we can do this consistently for all annotations for which it would make sense (via a util in the `annotations` package).

But given we already have a lot of annotations like this, I can't remember users complaining that setting it to false lead to unexpected behavior.

Reply:
We’ve had instances where customers have previously had annotation flags like this committed into their gitops repos and instead of removing the annotation they change the value to false to clearly indicate the difference to the previous configuration. This has caused unexpected behaviour.

Reply:
I agree a consistent handling in the annotation package is best

Reply:
Done, PTAL!

Reply:
👍  I like it!